lib/window-utils.js

```diff
@@ -147,6 +155,7 @@ function destroy(isUnload) {
   closeWindow();
   if (isUnload) {
     Services.obs.removeObserver(onWindowClosed, 'xul-window-destroyed', false);
+    Services.obs.removeObserver(closeRequested, 'quit-application-granted', false);
```

Comment:
@meandavejustice I missed this, but you're removing the wrong message when exiting. I'll open a PR quickly to fix.